### ISSUE TITLE
Loading code with Metacello/Iceberg crashes on Windows 10 1903

### ISSUE THREAD

**Issue by girba** [guillep/pharo-core#4437]
The way to reproduce on Windows 10 1903:
- first, make sure the version is at `1903`
- take a Pharo7 image
- load a Metacello script from GitHub, such as:
```
EpMonitor current disable.
[ 
  Metacello new
    baseline: 'GToolkit';
    repository: 'github://feenkcom/gtoolkit/src';
    load
] ensure: [ EpMonitor current enable ].
```

==> The crash appears instantaneously

There are no `crash.dmp` which suggests that the error appears in an external library.

This bug should be treated as critical given that it basically renders Pharo 7 useless on Windows 10.

**Comment by peteruhnak** [comment]
btw this is the same issue as these

https://github.com/pharo-project/pharo/issues/3418 
https://github.com/pharo-project/opensmalltalk-vm/issues/8
https://github.com/OpenSmalltalk/opensmalltalk-vm/issues/416 

> There are no crash.dmp which suggests that the error appears in an external library.

you should be able to see the problematic library in Windows system events ( https://github.com/pharo-project/pharo/issues/3418#issuecomment-523131189 )

> This bug should be treated as critical given that it basically renders Pharo 7 useless on Windows 10.

There is a workaround to tell iceberg to use https instead of ssh as it is a libssh issue. Or use 32bit Pharo 7.

But neither option is always available to everyone (ssh-only repo servers, having dependencies on 64bit-only libs (such as gtoolkit))

**Comment by fuhrmanator** [comment]
I started the upgrade of my machine at home to 1903 last night and will experiment with this soon. 

Iceberg doesn't work with `https` if you're using 2FA on Github (technically there is a workaround with an application key, but it's a pain to type it in to Pharo each time). 

Perhaps this crash doesn't happen in WSL (another possible workaround for Windows 10).

**Comment by bencoman** [comment]
I believe this can be closed....
   [ANN] New Windows VM - Fixes 1903 error
   http://forum.world.st/ANN-New-Windows-VM-Fixes-1903-error-td5104127.html

<blockquote><div><strong><a href="http://forum.world.st/ANN-New-Windows-VM-Fixes-1903-error-td5104127.html">Pharo Smalltalk Users - [ANN] New Windows VM - Fixes 1903 error</a></strong></div><div>[ANN] New Windows VM - Fixes 1903 error. Hello, a new stable VM has been deployed. This VM uses a new version of libSSH allowing us to work in the latest Windows version. It can be directly...</div></blockquote>

**Comment by fuhrmanator** [comment]
This is now happening in the 1909 version of Windows 10 with the latest Launcher and Pharo 8 image. Yesterday, I downloaded the Pharo Launcher from Pharo.org; the About box says:


>Pharo Launcher version: 1.9.2
>
>Pharo Launcher is a tool that eases the management of Pharo images and their associated Virtual Machines. It can download official Pharo images from the Pharo repository, fetch the right virtual machine to run your image and many other things.  
>If you find a bug, please open an issue at https://github.com/pharo-project/pharo-launcher/issues.

I'm using a Pharo 8 image, the one listed as `Pharo 8.0 - 64bit (stable)` in this version of Launcher.

I try to load code using, e.g., 

```smalltalk
Metacello new
  repository: 'github://juliendelplanque/MagicLiteralsSelector/src';
  baseline: 'MagicLiteralsSelector';
  load
```

Pharo crashes. There is no feedback in the user interface except that the window just silently disappears. Here is the application error in Windows 10 Event Viewer:

```
Faulting application name: Pharo.exe, version: 5.0.0.0, time stamp: 0x00000000
Faulting module name: libssh2-1.dll, version: 0.0.0.0, time stamp: 0xaec0ae70
Exception code: 0xc0000005
Fault offset: 0x000000000001e6cb
Faulting process id: 0x7dc
Faulting application start time: 0x01d5e3914919dc48
Faulting application path: C:\Users\Cris\Documents\Pharo\vms\80-x64\Pharo.exe
Faulting module path: C:\Users\Cris\Documents\Pharo\vms\80-x64\libssh2-1.dll
Report Id: cd70328a-b51d-43df-b96f-d1f618439d85
Faulting package full name: 
Faulting package-relative application ID: 
```

Here's the result of the `winver` command:

![image](https://user-images.githubusercontent.com/7606540/74591198-c933b100-4fe3-11ea-8428-41b6b04ab431.png)

And info from the command line:

```
C:\>systeminfo | findstr /B /C:"OS Name" /C:"OS Version"
OS Name:                   Microsoft Windows 10 Pro
OS Version:                10.0.18363 N/A Build 18363
```

**Comment by fuhrmanator** [comment]
A workaround on my issue was to use the **Settings > Tools > Software Configuration Management > Iceberg > Enable Metacello integration > Remote type > `HTTPS (https://github.com/<username>/<project>.git)`** option. (I had to remove the `pharo-local` folder after the first crash). 

Edit: I think you can set this by evaluating: 

```smalltalk
Iceberg remoteTypeSelector: #httpsUrl 
```

**Comment by msrocka** [comment]
I can also confirm that this bug still exists in the current Pharo 8 VM on Windows 10, 64 bit. The strange thing is that I get this error on one machine but not on another one. The workaround of @fuhrmanator  works => thank you!

As above, it crashes in the `libssh2-1.dll` of Pharo:

```
Faulting application name: Pharo.exe, version: 5.0.0.0, time stamp: 0x00000000
Faulting module name: libssh2-1.dll, version: 0.0.0.0, time stamp: 0xaec0ae70
Exception code: 0xc0000005
Fault offset: 0x000000000001e6cb
Faulting process id: 0x45e0
Faulting application start time: 0x01d5f07903e361df
Faulting application path: C:\Users\ms\Documents\Pharo\vms\80-x64\Pharo.exe
Faulting module path: C:\Users\ms\Documents\Pharo\vms\80-x64\libssh2-1.dll
Report Id: dd647c2f-d1f8-41ad-8077-8b16a9dbfce5
Faulting package full name: 
Faulting package-relative application ID: 
```

VM details:

```
C:\Users\ms\Documents\Pharo\vms\80-x64\Pharo.exe
CoInterpreter VMMaker.oscog-eem.2509 uuid: 91e81f64-95de-4914-a960-8f842be3a194 Feb  7 2019
StackToRegisterMappingCogit VMMaker.oscog-eem.2509 uuid: 91e81f64-95de-4914-a960-8f842be3a194 Feb  7 2019
VM: 201902062351 https://github.com/OpenSmalltalk/opensmalltalk-vm.git Date: Wed Feb 6 15:51:18 2019 CommitHash: a838346 Plugins: 201902062351 https://github.com/OpenSmalltalk/opensmalltalk-vm.git

Win32 built on Feb  7 2019 00:22:45 GMT Compiler: 4.2.1 Compatible Clang 5.0.1 (tags/RELEASE_501/final)
VMMaker versionString VM: 201902062351 https://github.com/OpenSmalltalk/opensmalltalk-vm.git Date: Wed Feb 6 15:51:18 2019 CommitHash: a838346 Plugins: 201902062351 https://github.com/OpenSmalltalk/opensmalltalk-vm.git
CoInterpreter VMMaker.oscog-eem.2509 uuid: 91e81f64-95de-4914-a960-8f842be3a194 Feb  7 2019
StackToRegisterMappingCogit VMMaker.oscog-eem.2509 uuid: 91e81f64-95de-4914-a960-8f842be3a194 Feb  7 2019
```

**edit**: it is confusing: when I update the VM via the Pharo launcher I get the VM version from `Jan  6 2020` and this version does not crash:

```
:\Users\ms\Documents\Pharo\vms\80-x64\Pharo.exe
CoInterpreter * VMMaker-CompatibleUserName.1577968628 uuid: d729bf66-2ea7-5a6d-b1eb-fc005785636d Jan  6 2020
StackToRegisterMappingCogit * VMMaker-CompatibleUserName.1577968628 uuid: d729bf66-2ea7-5a6d-b1eb-fc005785636d Jan  6 2020
VM: 201911290039-dev-7 admin@MacBook-Pro-de-admin.local:dev/Pharo/git-repos/pharo-vm/repo Date: Thu Nov 28 21:39:57 2019 CommitHash: 9b6bd16d3 Plugins: 201911290039-dev-7 admin@MacBook-Pro-de-admin.local:dev/Pharo/git-repos/pharo-vm/repo

Win32 built on Jan  6 2020 16:23:32 CET Compiler: 4.2.1 Compatible Clang 8.0.1 (tags/RELEASE_801/final)
VMMaker versionString VM: 201911290039-dev-7 admin@MacBook-Pro-de-admin.local:dev/Pharo/git-repos/pharo-vm/repo Date: Thu Nov 28 21:39:57 2019 CommitHash: 9b6bd16d3 Plugins: 201911290039-dev-7 admin@MacBook-Pro-de-admin.local:dev/Pharo/git-repos/pharo-vm/repo
CoInterpreter * VMMaker-CompatibleUserName.1577968628 uuid: d729bf66-2ea7-5a6d-b1eb-fc005785636d Jan  6 2020
StackToRegisterMappingCogit * VMMaker-CompatibleUserName.1577968628 uuid: d729bf66-2ea7-5a6d-b1eb-fc005785636d Jan  6 2020
```

![image](https://user-images.githubusercontent.com/3520062/75677299-6866d280-5c8b-11ea-9f50-1812e3aef2c3.png)


However, when I download the VM from https://files.pharo.org/get-files/80/pharo64-win-stable.zip
it contains the `Feb  7 2019` version (this is also the version that is linked on the download site). The latest version from https://files.pharo.org/get-files/80/pharo64-win-latest.zip contains the `Jan  6 2020` version.

**Comment by fuhrmanator** [comment]
I had another crash today in Windows 10 with the same issue (trying to load projects). I'm using the latest Pharo Launcher (2.x).

The solution I found was to:

- quit all my running Pharo Images, 
- In Pharo Launcher, click the **VMs** button: 
![image](https://user-images.githubusercontent.com/7606540/103178041-78205a80-484d-11eb-8bdd-40f1da310b30.png)
- click the **Update** button.

For some reason, the VM that was running before was old (Feb 2020, I think - I didn't take a screen shot).

**Comment by Ducasse** [comment]
Hi chris we are on vacation this week.
So with the latest VM is the problem solved?

**Comment by fuhrmanator** [comment]
Hi Steph - bonnes vacances :) Yes, it's fine with the latest VM - it's why I didn't re-open. I just posted so if it happens again I can find this :) since I'm not doing Pharo every day. 

I updated my Pharo Launcher last week, so I took for granted that the VMs were the latest, too. But I understand now that's not the case, and I also added a feature request to PL for indicating when VMs are out of date on the local system.
